### PR TITLE
Moved MethodBody.cs to shared

### PR DESCRIPTION
**Pull request by Anipik**

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Reflection/MethodBody.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace System.Reflection
+{
+    public class MethodBody
+    {
+        protected MethodBody() { }
+
+        private byte[] _IL = null;
+        private ExceptionHandlingClause[] _exceptionHandlingClauses = null;
+        private LocalVariableInfo[] _localVariables = null;
+#if CORECLR
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Reflection/MethodBody.cs]: I am wondering whether we should split all these related types to pure abstract shared base class, and internal RuntimeXXX class. Similar to how we have done RuntimeLocalVariableInfo.

**Comment by jkotas** [comment]
There is a new warning during corelib now:
`src\System\Reflection\RuntimeLocalVariableInfo.cs(15,19): warning CS0628: 'RuntimeLocalVariableInfo.RuntimeLocalVariableInfo()': new protected member declared in sealed class [D:\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]`

You may fix it while you are on it.

File: src/System.Private.CoreLib/shared/System/Reflection/MethodBody.cs (hunk 1 above)
Comment (Anipik) [reply src/System.Private.CoreLib/shared/System/Reflection/MethodBody.cs]: I also think that its really good idea. The code will be much clear and binding code(attached) will be in src
Let me know if that's the final decision ?

Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/Reflection/MethodBody.cs]: Yes, I like it too.

**Comment by Anipik** [comment]
@jkotas  i am closing this one as this type is linked with ```ExceptionHandlingClause``` i will move both of them to shared in a single PR

**Pull request closed without merging**